packages/common/driver-definitions/src/storage.ts

```diff
@@ -450,3 +453,42 @@ export enum FetchSource {
 	default = "default",
 	noCache = "noCache",
 }
+
+/**
+ * This represents the snapshot tree structure with group IDs. It could be a partial snapshot tree
+ * where the missing tree structure inside another tree could be represented with a groupId and the
+ * actual tree structure could be missing. This group Ids could be then be used to fetch the actual
+ * tree structure from the storage service.
+ * @alpha
+ */
+export interface ISnapshotTree2 extends ISnapshotTree {
+	/**
+	 * Represents the group to which the children belongs to.
+	 */
+	groupID?: string;
+
+	trees: { [path: string]: ISnapshotTree2 };
+}
+
+/**
+ * This represents the complete snapshot along with contents. The snapshotTree in this structure
+ * could be a partial snapshot tree consisting of group Ids for the missing children of the snapshotTree.
+ * @alpha
+ */
+export interface IPartialSnapshotWithContents {
+	snapshotTree: ISnapshotTree2;
+	blobsContents: Map<string, ArrayBuffer>;
```

Comment:
I do not know if it's a good direction, but worth pointing out that
1. If you make IPartialSnapshotWithContents castable to ISnapshotTree(2)
2. Then possibly you could stop adding IPartialSnapshotWithContents to all existing APIs.

It would carry less info (for developers), but it would be correct from type perspective.
Any place where you would test for if it's IPartialSnapshotWithContents, compiler would immediately know (through instanceOfIPartialSnapshotWithContents signature) that you deal with IPartialSnapshotWithContents type.

Reply:
Done.